src/lib.rs

```diff
@@ -626,7 +690,14 @@ pub fn list() -> Vec<(String, String)> {
 #[doc(hidden)]
 pub fn eval<R, F: FnOnce(Option<String>) -> R>(name: &str, f: F) -> Option<R> {
     let p = {
-        let registry = REGISTRY.registry.read().unwrap();
+        let registry = {
+            let group = REGISTRY_GROUP.read().unwrap();
+            let id = thread::current().id();
+            group.get(&id).unwrap_or(&REGISTRY_GLOBAL.registry).clone()
+        };
+
+        let registry = registry.read().unwrap();
+
         match registry.get(name) {
             None => return None,
             Some(p) => p.clone(),
```

Comment:
How about find the fail point at the local registry first, if there are not local registry or not such fail point with `name` at the local registry, then try to find it at the global registry.

Reply:
Let's get a consensus on this proposal @NingLin-P @BusyJay